src/wallet/scriptpubkeyman.cpp

```diff
@@ -624,7 +624,7 @@ TransactionError LegacyScriptPubKeyMan::FillPSBT(PartiallySignedTransaction& psb
         }
 
         // Get the Sighash type
-        if (sign && input.sighash_type > 0 && input.sighash_type != sighash_type) {
+        if (sign && input.sighash_type > 0 && sighash_type > 0 && input.sighash_type != sighash_type) {
```

Comment:
Is this right? I think we should distinguish based on whether the PSBT having having a PSBT_IN_SIGHASH_TYPE record and not. I think the current code treats there not being a record, and the record being 0 as identical?

Reply:
Hmm, yes. I've changed `input.sighash_type` to be an optional. The issue this particular change was test related, so I've updated the tests instead.